### PR TITLE
feat: adds the ability to set the index strategy

### PR DESCRIPTION
**Pull request by tdejager**
This PR enables to set the UV `index-strategy` from pixi. I would like a test for this so it is still in draft for now :)

closes: #1377

### CODE REVIEW

**Review by baszalmstra — APPROVED** [review]

File: pixi.toml [hunk 1]
```diff
@@ -115,6 +122,7 @@ docs = { features = [
 lint = { features = [
   "lint",
 ], no-default-feature = true, solve-group = "default" }
+pypi-gen = { features = ["pypi-gen"] }
```
Comment (Hofer-Julian) [inline pixi.toml:125]: Would it make sense to add solve_group "default" here?

File: tests/pypi-indexes/generate-indexes.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+"""This is a little script to generate a custom pypi simple index from a directory of source packages."""
+
+from pathlib import Path
+import shutil
+from build import ProjectBuilder
+
+indexes_path = Path(__file__).parent
+
+index_html_template = """<!DOCTYPE html>
```
Comment (Hofer-Julian) [inline tests/pypi-indexes/generate-indexes.py:9]:
```suggestion
index_html_template = """\
<!-- This file has been generated with `generate-indexes.py` --> 
<!DOCTYPE html>
```

When running this script again, this should make it obvious where these files come from

File: crates/pixi_manifest/src/pypi/pypi_options.rs [hunk 3]
```diff
@@ -93,6 +123,20 @@ impl PypiOptions {
             self.index_url.clone()
         };
 
+        // Allow only one index strategy
+        let index_strategy = if let Some(other_index_strategy) = other.index_strategy.clone() {
+            if let Some(own_index_strategy) = self.index_strategy.clone() {
+                return Err(PypiOptionsMergeError::MultipleIndexStrategies {
+                    first: own_index_strategy.to_string(),
+                    second: other_index_strategy.to_string(),
+                });
```
Comment (Hofer-Julian) [inline crates/pixi_manifest/src/pypi/pypi_options.rs:132]:
```suggestion
            if let Some(ref own_index_strategy) = self.index_strategy {
                return Err(PypiOptionsMergeError::MultipleIndexStrategies {
                    first: own_index_strategy.to_string(),
                    second: other_index_strategy.to_string(),
                });
```

Something like this?
I always forget the syntax.

**Review by Hofer-Julian — COMMENTED** [review]
The code looks good to me. Ony had minor comments

File: pixi.toml (hunk 1 above)
Comment (baszalmstra) [reply pixi.toml:125]: What would be the reason to do that? Its just a script to generate some files so it doesnt need to share any of the dependencies with other environments.

File: crates/pixi_manifest/src/pypi/pypi_options.rs (hunk 3 above)
Comment (baszalmstra) [reply crates/pixi_manifest/src/pypi/pypi_options.rs:132]:
```suggestion
            if let Some(own_index_strategy) = &self.index_strategy {
                return Err(PypiOptionsMergeError::MultipleIndexStrategies {
                    first: own_index_strategy.to_string(),
                    second: other_index_strategy.to_string(),
                });
```

**Comment by tdejager** [comment]
BTW I forgot, but before merging we need to change the schema and add docs.

File: pixi.toml (hunk 1 above)
Comment (Hofer-Julian) [reply pixi.toml:125]: Fair enough.

**Comment by Hofer-Julian** [comment]
> BTW I forgot, but before merging we need to change the schema and add docs.

Unrelated to this PR, but I already wondered why we generate the schema from pydantic instead from our Rust struct. I assume that's not easy to do with serde?

**Comment by tdejager** [comment]
> > BTW I forgot, but before merging we need to change the schema and add docs.
> 
> Unrelated to this PR, but I already wondered why we generate the schema from pydantic instead from our Rust struct. I assume that's not easy to do with serde?

I think the last time we checked that crate was less expressive?

**Comment by tdejager** [comment]
> BTW I forgot, but before merging we need to change the schema and add docs.

Done.

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**